postgres/pgdump.go

```diff
@@ -470,18 +474,52 @@ func extractConstraints(conv *internal.Conv, n nodes.Node, table string, l []nod
 	for _, i := range l {
 		switch d := i.(type) {
 		case nodes.Constraint:
-			var cols []string
-			for _, j := range d.Keys.Items {
-				k, err := getString(j)
+			var cols, referCols []string
+			var referTable string
+			switch d.Contype {
+			case nodes.CONSTR_FOREIGN:
+				t, err := getTableName(conv, *d.Pktable)
 				if err == nil {
-					cols = append(cols, k)
+					referTable = t
 				}
 				if err != nil {
 					conv.Unexpected(fmt.Sprintf("Processing %v statement: error processing constraints: %s", reflect.TypeOf(n), err.Error()))
 					conv.ErrorInStatement(prNodes([]nodes.Node{n, d}))
+					continue
+				}
+				for i := range d.FkAttrs.Items {
+					k, err := getString(d.FkAttrs.Items[i])
+					if err == nil {
```

Comment:
This was already in issue in the original code.

But either this should be if-else.
Or maybe more idiomatic as:

if err != nil {
   conv.Unexpected(fmt.Sprintf("Processing %v statement: error processing constraints: %s", reflect.TypeOf(n), err.Error()))
  conv.ErrorInStatement(prNodes([]nodes.Node{n, d}))
  continue
}
cols = append(cols, k)

Reply:
Got it, I have converted it.